nucypher/blockchain/eth/registry.py

```diff
@@ -374,17 +379,14 @@ def search(self, beneficiary_address: str = None, contract_address: str = None):
             else:
                 contract_data = records[0]
 
-        else:
-            raise ValueError("Beneficiary address or contract address must be supplied.")
-
         return contract_data
 
     def enroll(self, beneficiary_address, contract_address, contract_abi) -> None:
         contract_data = [contract_address, contract_abi]
         try:
             allocation_data = self.read()
         except self.RegistryError:
-            self.log.info("Blank allocation registry encountered: enrolling {}:{}".format(beneficiary_address, contract_address))
+            self.log.info(f"Blank allocation registry encountered: enrolling {beneficiary_address}:{contract_address}")
```

Comment:
```suggestion
            self.log.info(f"Blank allocation registry encountered when enrolling {beneficiary_address}:{contract_address}")
```

Reply:
^ does `RegistryError` only occur if blank - is it possible there is something but the format is incorrect?

Reply:
Yeah, it's possible since were catching a base class that another condition has occurred.  Needs an eye for review - The addition of `class BlankRegistry(RegistryError)` will help to alleviate this.

Reply:
Apart from the code suggestion, not sure how to proceed here.
@KPrasch are you suggesting to add `class BlankRegistry(RegistryError)` now?

Reply:
My take would be to update the message to be more general. `Blank allocation` doesn't seem to be the only condition for the Error. If you want to differentiate between Blank registry and some other error scenario, then you can do the subclass thing.